### PR TITLE
Add Tests for QuadKey and VenueService

### PR DESCRIPTION
**Pull request by ckskate**

### CODE REVIEW

**Review by abdullahselek — CHANGES_REQUESTED** [review]
Separate commits would be better actually, I will add my comments for each commit

File: OnDemandPassenger/VenueMapTileLayerUtility.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//
+//  VenueMapTileLayerUtility.h
+//  OnDemandPassenger
+//
+//  Created by Killion, Connor on 5/23/17.
+//  Copyright © 2017 HERE. All rights reserved.
+//
```
Comment (abdullahselek) [inline OnDemandPassenger/VenueMapTileLayerUtility.h]: You can change copyright header we should all use the same one.
```
/*
 * Copyright (c) 2017 HERE Europe B.V.
 * All rights reserved.
 */
```

File: OnDemandPassenger/VenueMapTileLayerUtility.m [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//
+//  VenueMapTileLayerUtility.m
+//  OnDemandPassenger
+//
+//  Created by Killion, Connor on 5/23/17.
+//  Copyright © 2017 HERE. All rights reserved.
+//
```
Comment (abdullahselek) [inline OnDemandPassenger/VenueMapTileLayerUtility.m]: Copyright header

File: OnDemandPassengerTests/VenueMapTileLayerTests.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2017 HERE Europe B.V.
+ * All rights reserved.
+ */
+
+import XCTest
+
+class VenueMapTileLayerTests: XCTestCase {
```
Comment (abdullahselek) [inline OnDemandPassengerTests/VenueMapTileLayerTests.swift]: We use Quick framework for our unit tests you can get ViewControllerTests as an example.

File: OnDemandPassengerTests/OnDemandPassengerTests-Bridging-Header.h [hunk 4]
```diff
@@ -0,0 +1,9 @@
+//
```
Comment (abdullahselek) [inline OnDemandPassengerTests/OnDemandPassengerTests-Bridging-Header.h]: Why did you add this file? Bridging header is used for reaching Objective-C classes from Swift classes. Our project is already based on Objective-C.

File: OnDemandPassengerTests/VenueServiceTests.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2017 HERE Europe B.V.
+ * All rights reserved.
+ */
+
+import XCTest
+
+class VenueServiceTests: XCTestCase {
```
Comment (abdullahselek) [inline OnDemandPassengerTests/VenueServiceTests.swift]: We use Quick framework for our unit tests you can get ViewControllerTests as an example.

File: OnDemandPassengerUITests/OnDemandPassengerUITests.swift [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+//  OnDemandPassengerUITests.swift
+//  OnDemandPassengerUITests
+//
+//  Created by Killion, Connor on 5/26/17.
+//  Copyright © 2017 HERE. All rights reserved.
+//
+
+import XCTest
+
+class OnDemandPassengerUITests: XCTestCase {
```
Comment (abdullahselek) [inline OnDemandPassengerUITests/OnDemandPassengerUITests.swift]: We use Quick framework for our unit tests you can get ViewControllerTests as an example.

**Review by abdullahselek — CHANGES_REQUESTED** [review]
I added a few comments take a look.

File: OnDemandPassenger/Constants.m [hunk 7]
```diff
@@ -6,6 +6,6 @@
 #import "Constants.h"
 
 #warning Replace with your app id
-NSString * APP_ID = @"YOUR_APP_ID";
```
Comment (SirBif) [inline OnDemandPassenger/Constants.m]: Please remove these from this commit. I will invalidate them now, but make sure not to commit credentials

File: OnDemandPassengerTests/VenueMapTileLayerTests.swift [hunk 8]
```diff
@@ -24,9 +24,16 @@ class VenueMapTileLayerTests: XCTestCase {
   
   // test edge case of 0 as input.
   func testQuadkeyGenerationZero() {
```
Comment (SirBif) [inline OnDemandPassengerTests/VenueMapTileLayerTests.swift]: Can you squash/fixup this change with the previous commit? It's cleaner to have just a single commit for this change

File: OnDemandPassengerTests/VenueServiceTests.swift [hunk 9]
```diff
@@ -6,16 +6,26 @@
...
   
   override func setUp() {
     super.setUp()
     
-    let storyboard = UIStoryboard(name: "Main", bundle: Bundle.main)
-    vc = storyboard.instantiateViewController(withIdentifier: "SearchViewController") as! SearchViewController
+    sharedVenueService = VenueService()
   }
   
   func testPlacesRequestTriggeredProperly() {
```
Comment (SirBif) [inline OnDemandPassengerTests/VenueServiceTests.swift]: The name does not match the test. This is checking that the venue service requests the signature, right?

File: OnDemandPassenger/Constants.m [hunk 10]
```diff
@@ -6,6 +6,6 @@
 #import "Constants.h"
 
 #warning Replace with your app id
-NSString * APP_ID = @"TQuyOFX2nNRkFGjjVKDy";
```
Comment (SirBif) [inline OnDemandPassenger/Constants.m]: You don't need this commit, you should remove the credentials from the first one

File: OnDemandPassengerTests/VenueServiceTests.swift [hunk 11]
```diff
@@ -19,7 +19,6 @@ class VenueServiceTests: XCTestCase {
     let expect = expectation(description: "VenueService will receive a signature and store it once")
     
     sharedVenueService?.requestSignature(DispatchQueue.main) { (signedQueryString: String?) -> Void in
-      XCTAssert(signedQueryString != nil)
```
Comment (SirBif) [inline OnDemandPassengerTests/VenueServiceTests.swift]: This change should be part of the venue service test commit

File: OnDemandPassengerUITests/OnDemandPassengerUITests.swift [hunk 12]
```diff
@@ -38,14 +38,11 @@ class OnDemandPassengerUITests: XCTestCase {
     app.children(matching: .window).element(boundBy: 0).children(matching: .other).element.children(matching: .other).element.children(matching: .other).element.children(matching: .other).element.tap()
   }
   
-  func testSearchViewControllerViewIsUpdatedProperly() {
```
Comment (SirBif) [inline OnDemandPassengerUITests/OnDemandPassengerUITests.swift]: This change should be part of the search view controller test commit

**Review by SirBif — CHANGES_REQUESTED** [review]
Please clean up the commit history so there's only one commit per topic (e.g. one to add the test for the quadkey, one for the search view controller). And remove the merge commit, try rebasing on top of master instead

File: OnDemandPassenger/Constants.m (hunk 7 above)
Comment (SirBif) [reply OnDemandPassenger/Constants.m]: Mmm.. I see my comments don't show up nicely in the unified view. My point was to amend the first commit to remove the credentials instead of adding a new commit at the end to delete them.

File: OnDemandPassengerUITests/OnDemandPassengerUITests.swift (hunk 12 above)
Comment (SirBif) [reply OnDemandPassengerUITests/OnDemandPassengerUITests.swift]: removed these tests. the wouldn't run without credentials

**Review by SirBif — DISMISSED** [review]

**Comment by SirBif** [comment]
Does it look good @sschuberth ?

**Comment by sschuberth** [comment]
Great @SirBif, LGTM! Please go ahead and merge.

**Review by SirBif — APPROVED** [review]
Approving after @sschuberth review
